keps/sig-node/4800-cpumanager-split-uncorecache/README.md

```diff
@@ -0,0 +1,689 @@
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+We plan on adding/modifying functions to the following files to create the uncorecache alignment feature:
+- pkg/kubelet/cm/cpumanager/cpu_assignment.go
+- /pkg/kubelet/cm/cpumanager/cpu_manager.go
+- /pkg/kubelet/cm/cpumanager/policy_options.go
+- /pkg/kubelet/cm/cpumanager/topology/topology.go
+
+Existing topoology test cases will be modified to include uncorecache topology. All modified and added functions will have new test cases.
+
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+- `/pkg/kubelet/cm/cpumanager/cpu_assignment.go`: `2024-08-26` - `93.5%`
+- `/pkg/kubelet/cm/cpumanager/cpu_manager.go`: `2024-08-26` - `74.5%`
+- `/pkg/kubelet/cm/cpumanager/policy_static.go`: `2024-08-26` - `89.5%`
+- `/pkg/kubelet/cm/cpumanager/topology/topology.go`: `2024-9-30` - `93.2%`
+
+##### Integration tests
+
+N/A. This feature requires a e2e test for testing.
+
+##### e2e tests
+
+- For e2e testing, checks will be added to determine if the node has a split uncore cache topology. If node does not meet the requirement to have multiple uncore caches, the added tests will be skipped. 
+- e2e testing should cover the deployment of a pod that is following uncore cache alignment. CPU assignment can be determined by podresources API and programatically cross-referenced to syfs topology information to determine proper uncore cache alignment.
+- For e2e testing, guaranteed pods will be deployed with various CPU size requirements on our own baremetal instances across different vendor architectures and confirming the CPU assignments to uncore cache core groupings. This feature is intended for baremetal only and not cloud instances.
+- Update CI to test GCP instances of different architectures utilizing uncore cache alignment feature.
+
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature gate flag option
+- Initial e2e tests completed and enabled
```

Comment:
the caveat could be the tests would skip most of the time until we fix the provisioning of CI hardware (see ongoing https://github.com/kubernetes/k8s.io/issues/7339), but should not be a problem for alpha

Reply:
Let's add it here explicitly, so we don't forget.

Reply:
Would that go here in the graduation criteria or in the caveat section?

Reply:
added